cam_diagnostics/cam_diagnostics.F90

```diff
@@ -0,0 +1,211 @@
+      call history_out_field('U'        , u)
+      call history_out_field('V'        , v)
+      call history_out_field('DSE'      , dse)
+      call history_out_field('OMEGA'    , omega)
+      call history_out_field('PMID'     , pmid)
+      call history_out_field('PMIDDRY'  , pmiddry)
+      call history_out_field('PDEL'     , pdel)
+      call history_out_field('PDELDRY'  , pdeldry)
+      call history_out_field('RPDEL'    , rpdel)
+      call history_out_field('RPDELDRY' , rpdeldry)
+      call history_out_field('LNPMID'   , lnpmid)
+      call history_out_field('LNPMIDDRY', lnpmiddry)
+      call history_out_field('EXNER'    , inv_exner)
+      call history_out_field('ZM'       , zm)
+      call history_out_field('PINT'     , pint)
+      call history_out_field('PINTDRY'  , pintdry)
+      call history_out_field('LNPINT'   , lnpint)
+      call history_out_field('LNPINTDRY', lnpintdry)
+      call history_out_field('ZI'       , zi)
+
+      ! Capture constituent fields
+      const_num_found = 0
+      do const_idx = 1, size(const_props)
+         call const_props(const_idx)%standard_name(standard_name, errflg, errmsg)
+         do name_idx = 1, size(const_std_names)
+            if (trim(standard_name) == trim(const_std_names(name_idx))) then
+               call history_out_field(trim(const_diag_names(name_idx)), const_array(:,:,const_idx))
+               const_num_found = const_num_found + 1
+            end if
+         end do
+         if (const_num_found == size(const_std_names)) then
+            exit
+         end if
+      end do
+
+   end subroutine cam_state_diagnostics_run
+
+   !> \section arg_table_cam_tend_diagnostics_init  Argument Table
+   !! \htmlinclude cam_tend_diagnostics_init.html
+   subroutine cam_tend_diagnostics_init(errmsg, errflg)
+      use cam_history, only: history_add_field
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      ! Add tendency fields
+      call history_add_field('TTEND', 'tendency_of_air_temperature_due_to_model_physics', 'lev', 'avg', 'K s-1')
+      call history_add_field('UTEND', 'tendency_of_eastward_wind_due_to_model_physics',   'lev', 'avg', 'm s-2')
+      call history_add_field('VTEND', 'tendency_of_northward_wind_due_to_model_physics',  'lev', 'avg', 'm s-2')
+
+   end subroutine cam_tend_diagnostics_init
+
+   !> \section arg_table_cam_tend_diagnostics_run  Argument Table
+   !! \htmlinclude cam_tend_diagnostics_run.html
+   subroutine cam_tend_diagnostics_run(dTdt_total, dudt_total, dvdt_total, errmsg, errflg)
+      use cam_history, only: history_out_field
+      ! Tendency variables
+      real(kind_phys), intent(in) :: dTdt_total(:,:) ! tendency of air temperature due to model physics
+      real(kind_phys), intent(in) :: dudt_total(:,:) ! tendency of eastward wind due to model physics
+      real(kind_phys), intent(in) :: dvdt_total(:,:) ! tendency of northward wind due to model physics
```

Comment:
Cam has flx_net, te_tnd and tw_tnd as wll as the above variables in the physics_tend structure.  Should these be included?

Reply:
made issue https://github.com/ESCOMP/atmospheric_physics/issues/128